spring-cloud-deployer-local/src/main/java/org/springframework/cloud/deployer/spi/local/LocalTaskLauncher.java

```diff
@@ -234,38 +230,17 @@ public void shutdown() throws Exception {
 		taskInstanceHistory.clear();
 	}
 
-	private Path createWorkingDir(String taskLaunchId, Path dir) throws IOException {
+	private Path createWorkingDir(Map<String, String> deploymentProperties, String taskLaunchId) throws IOException {
+		LocalDeployerProperties localDeployerPropertiesToUse = bindDeploymentProperties(deploymentProperties);
 
-		Path workDir = Files.createDirectory(Paths.get(dir.toFile().getAbsolutePath(),
-				taskLaunchId));
-
-		if (getLocalDeployerProperties().isDeleteFilesOnExit()) {
+		Path workingDirectoryRoot = Files.createDirectories(localDeployerPropertiesToUse.getWorkingDirectoriesRoot());
+		Path workDir = Files.createDirectories(workingDirectoryRoot.resolve(Long.toString(System.nanoTime())).resolve(taskLaunchId));
+		if (localDeployerPropertiesToUse.isDeleteFilesOnExit()) {
 			workDir.toFile().deleteOnExit();
 		}
-
 		return workDir;
 	}
 
-	private Path createLogDir(AppDeploymentRequest request) throws IOException {
-
-		if (this.logPathRoot == null) {
-			this.logPathRoot =
-					Files.createTempDirectory(getLocalDeployerProperties().getWorkingDirectoriesRoot(),
```

Comment:
Yeah this is what broke that binding. `LocalTaskPlatformFactory` in dataflow server core did the binding. Using `getLocalDeployerProperties()` vs. deployment properties vs. how things are setup from scdf side needs to be fixed/aligned.